tuna/miopen/celery_tuning/tuning.py

```diff
@@ -290,102 +291,75 @@ def tune(library, job_batch_size=1000):
-          for job, config in serialized_jobs:
-            context = {
-                'job': job,
-                'config': config,
-                'worker_type': worker_type,
-                'arch': library.dbt.session.arch,
-                'num_cu': library.dbt.session.num_cu,
-                'kwargs': kwargs,
-                'fdb_attr': fdb_attr
-            }
-
-            #calling celery task, enqueuing to celery queue
-            res_set.add(
-                celery_enqueue.apply_async((context,),
-                                           queue=q_name,
-                                           reply_to=q_name))
-
-        if not job_list:
-          if not res_set:
-            results_gather_terminate(res_set, drain_process)
-            return False
-          LOGGER.info('All tasks added to queue')
-          j_end = time.time()
-          break
-      except KeyboardInterrupt:
-        LOGGER.error('Keyboard interrupt caught, draining results queue')
-        session.rollback()
-        results_gather_terminate(res_set, drain_process)
-        purge_queue([q_name])
-        library.cancel_consumer(q_name)
-
-  results_gather_terminate(res_set, drain_process)
-  library.cancel_consumer(q_name)
-  end = time.time()
-  LOGGER.info("Took {:0>8} to tune".format(str(timedelta(seconds=end - start))))  #pylint: disable=consider-using-f-string
-  LOGGER.info("{:0>8} of which was spent enqueuing jobs".format(  #pylint: disable=consider-using-f-string
-      str(timedelta(seconds=j_end - start))))
+    job_list = library.get_jobs(session, library.fetch_state,
+                                library.set_state, library.args.session_id, no_update=True)
+  job_counter = Value('i', len(job_list))
+  LOGGER.info('Job counter: %s', job_counter.value)
 
-  return True
-
-
-def results_gather_start(worker_type):
-  """Start subproc to drain result queue and populate mysql DB"""
-  LOGGER.info('Started drain process')
-  #start draining result_queue in subprocess
-  event = threading.Event()
-  drain_process = Process(target=drain_queue, args=[worker_type, DBT, event])
-  drain_process.start()
-
-  return drain_process
 
+  try:
+    enqueue_proc = Process(target=enqueue_jobs, args=[library, job_batch_size, worker_type, kwargs, fdb_attr, q_name])
+    #Start enqueue proc
+    enqueue_proc.start()
+    #start async consume thread, blocking
+    asyncio.run(consume(job_counter, worker_type, DBT))
+    enqueue_proc.join()
 
-def result_callback(task_id, value):
-  """Function callback for celery async jobs to store results"""
-  LOGGER.info('task id %s : done', task_id)
-  LOGGER.info('fin_json : %s', value[0])
-  LOGGER.info('context : %s', value[1])
-  #result, v = app.AsyncResult(task_id).get()
-  #LOGGER.info('v: %s', v)
-  #LOGGER.info('results: %s', result)
-
-  result_queue.put([value[0], value[1]])
+  except KeyboardInterrupt:
+    LOGGER.error('Keyboard interrupt caught, draining results queue')
+    purge_queue([q_name])
```

Comment:
If you dont purge the queue, you would not have to requeue when you restart. However, that might make state management more complicated.

Reply:
maybe once we have the ability to reset state on ctrl+c we can look into it.